src/app/specialties/specialty-edit/specialty-edit.component.html

```diff
@@ -16,25 +16,28 @@
   ~  */
   -->
 
-<div class="container-fluid">
-  <div class="container xd-container">
-    <h2>Edit Specialty</h2>
-    <form id="specialty" class="form-horizontal" (ngSubmit)="onSubmit(specialtyForm.value)" #specialtyForm="ngForm">
-      <div class="form-group" hidden="true">
-        <input type="text" hidden="true" class="form-control" id="id" [(ngModel)]="specialty.id" name="id"/>
-      </div>
-      <div class="form-group has-feedback">
-        <div class="form-group has-feedback" [class.has-success]="name.dirty && name.valid" [class.has-error]="name.dirty && !name.valid">
-          <label class="col-sm-1 control-label">Name</label>
-          <div class="col-sm-6">
-            <input id="name" name="name" class="form-control" type="text" required [(ngModel)]="specialty.name" #name="ngModel"/>
-            <span class="glyphicon form-control-feedback" [class.glyphicon-ok]="name.valid" [class.glyphicon-remove]="!name.valid" aria-hidden="true"></span>
-            <span class="help-block" *ngIf="name.dirty && name.hasError('required')">Name is required</span>
+  <div class="container-fluid">
+    <div class="container xd-container">
+      <h2>Edit Specialty</h2>
+      <form id="specialty" class="form-horizontal" (ngSubmit)="onSubmit(specialtyForm.value)" #specialtyForm="ngForm">
+        <div class="form-group" hidden="true">
+          <input type="text" hidden="true" class="form-control" id="id" [(ngModel)]="specialty.id" name="id"/>
+        </div>
+        <div class="form-group has-feedback">
+          <div class="form-group has-feedback" [class.has-success]="name.dirty && name.valid" [class.has-error]="name.dirty && !name.valid">
+            <label class="col-sm-1 control-label">Name</label>
+            <div class="col-sm-6">
+              <input id="name" name="name" class="form-control" type="text"  minlength="1" maxlength="30" pattern="^[A-Za-z0-9].{0,29}$" required [(ngModel)]="specialty.name" #name="ngModel"/>
```

Comment:
issue: name may be only 80 characters long
`maxlength="80" pattern="^[A-Za-z0-9].{0,79}$" `